pkg/arvo/lib/string.hoon

```diff
@@ -0,0 +1,561 @@
+  ::  /lib/string
+::::
+::  Implements a set of userspace functions for string manipulation adopting
+::  similar names to the Python string standard library.  All inputs are tapes.
+::
+::  This file is not subject to kelvin versioning and the interface should
+::  not be considered official.
+::
+|%
+::
+::  ++alphabet
+::
+::  26-letter Roman alphabet, upper case and lower case. 
+::
+::  Source: 
+++  alphabet        ^~  `tape`(weld (gulf 65 90) (gulf 97 122))
+::
+::  ++alpha-lower
+::
+::  26-letter Roman alphabet, lower case. 
+::
+::  Source: 
+++  alpha-lower     ^~  `tape`(slag 26 alphabet)
+::
+::  ++alpha-upper
+::
+::  26-letter Roman alphabet, lower case. 
+::
+::  Source: 
+++  alpha-upper     ^~  `tape`(scag 26 alphabet)
+::
+::  ++digits
+::
+::
+::  Ten decimal digits. 
+::
+::  Source: 
+++  digits          ^~  `tape`(gulf 48 57)
+::
+::  ++alpha-digits
+::
+::  26-letter Roman alphabet, upper case and lower case, plus ten decimal digits. 
+::
+::  Source: 
+++  alpha-digits    ^~  `tape`(weld alphabet digits)
+::
+::  ++hexdigits
+::
+::
+::  Sixteen hexadecimal digits, upper case and lower case. 
+::
+::  Source: 
+++  hexdigits       ^~  `tape`:(weld digits (gulf 65 70) (gulf 97 102))
+::
+::  ++octdigits
+::
+::  Eight octal digits. 
+::
+::  Source: 
+++  octdigits       ^~  `tape`(gulf 48 55)
+::
+::  ++punctuation     tape:(weld (gulf 32 47) (gulf 58 64) (gulf 91 96) (gulf 123 126))
+::
+::  All ASCII punctuation characters. 
+::
+::  Source: 
+++  punctuation     ^~  `tape`:(weld (gulf 32 47) (gulf 58 64) (gulf 91 96) (gulf 123 126))
+::
+::  ++whitespace      tape:(weld " " ~['a'] ~['9'])
+::
+::  All ASCII whitespace characters. 
+::
+::  Source: 
+++  whitespace      ^~  `tape`:(weld " " ~['a'] ~['9'])
```

Comment:
why are 'a' and '9' considered whitespace?

Reply:
do you mean 0xa and 0x9 ?

Reply:
Should be `~['\0a'] ~['\09']`